src/Container/AutowireFactory.php

```diff
@@ -17,13 +17,13 @@ class AutowireFactory
      * Retrieves the injector from a container
      *
      * @param ContainerInterface $container The container context for this factory
-     * @return InjectorInterface The dependency injector
      * @throws Exception\RuntimeException When no dependency injector is available.
      */
-    private function getInjector(ContainerInterface $container)
+    private function getInjector(ContainerInterface $container): InjectorInterface
     {
         $injector = $container->get(InjectorInterface::class);
 
+        /** @psalm-suppress DocblockTypeContradiction Can we remove all code below and return directly from container get? */
```

Comment:
As mentioned in another comment, just because you request `InjectorInterface::class` from the container, that **must** not be an instance of the requested class-string.

I would be fine with that assertion and also with crashing the application with a type-error if some upstream dev thinks he can hassle with the dependency injection but thats just my opinion about this but @Ocramius had a point on this.

Reply:
IMHO: We can assume that the container provides matching class instances, since this is what laminas-di provides: Autowire for Psr Contrainers when requesting `$container->get(Some::class)`.

I'd also argue that this can be handled by PHP's type checker rather than checking it manually - in the end it's less code to maintain.